container/responsive.go

```diff
@@ -0,0 +1,59 @@
+package container
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/x/fyne/layout"
+)
+
+// NewResponsive returns a container with a responsive layout. The objects
+// can be copmmon containers or responsive objects using the Responsive()
+// function. Note that the content size is computed from the container size and not
+// from the window size.
+//
+// Example:
+//
+//	NewResponsive(
+//		widget.NewLabel("Hello World"), // 100% by default
+//		Responsive(widget.NewLabel("Hello World"), 1, .5), // 100% for small, 50% for others
+//		Responsive(widget.NewLabel("Hello World"), 1, .5), // 100% for small, 50% for others
+//	)
+func NewResponsive(objects ...fyne.CanvasObject) *fyne.Container {
+	container := container.New(layout.NewResponsiveLayout())
+	container.Objects = objects
+	return container
+}
+
+// Responsive returns a responsive object configured with breakpoint sizes.
+// The number of ratios can be up to 5, for extra small, small, medium, large and extra large and above.
+// If no size is provided, the object will be 100% of the layout for the whole possible size breakpoints.
+// If more than 5 ratios are provided, the extra ratios are ignored.
+// If less than 5 raiios are provided, the last ratio is used for the missing sizes.
+//
+// This sizes are used for the following breakpoints:
+//
+//   - extra small: 0px to 576px
+//   - small: 567px to 767px
+//   - medium: 768px to 992 px
+//   - large: 992px to 1200
+//   - extra large: 1281px and above
+//
+// Example:
+//
+//	Responsive(widget.NewLabel("Hello World"), 1, .5)
+//
+// It's commonly use in a responsive container, like this:
+//
+//	NewResponsive(
+//		Responsive(widget.NewLabel("Hello World"), 1, .5),
+//		Responsive(widget.NewLabel("Hello World"), 1, .5, .25),
+//	)
+//
+// Or with the Add() method:
+//
+//	ctn := NewResponsive()
+//	ctn.Add(Responsive(widget.NewLabel("Hello World"), 1, .5))
+//	ctn.Add(Responsive(widget.NewLabel("Hello World"), 1, .5))
+func Responsive(object fyne.CanvasObject, ratios ...Ratio) fyne.CanvasObject {
```

Comment:
Hmm, is adding more static functions desirable to handle what is metadata for a single type of container?
It may be a wider design question but if the info cannot be passed in constructor is there a simpler way to handle it that keeps the API entry points to a lower number?

My concern is that "Responsive" could be assumed to work in other places. Like if we accidentally added a "StayLeft" function to describe the position of an item in a border container...

Reply:
It's a question I've been asking myself for some time.

`NewReponsive` returns a `*fyne.Container` where the `Add()` method only accepts `fyne.CanvasObject`, which is perfectly fine.

As a result, I have two possible choices:

- either I allow metadata to be added, as I do with `Reponsive()`, but this effectively transforms the objects into "something else with metadata". But the advantage is that I can always add them to another container without affecting the way it works (for example, to test, I change my container to Grid, which doesn't cause any problems).
- or I overload `fyne.Container` to return a `ResponsiveContainer` that have a `Add()` method that accepts responsive configuration. But then I'll have two problems:
    - if I change my container to test another one, the "Add()" method will no longer be respected
    - functions that accept `fyne.Container` will refuse my object

I was thinking of allowing this with `NewResponsive()` by returning a "config" function signed like this: `func(widget, ratios... float32)`.

So to do it this way:

```go
ctn, configFunc := NewResponsive(label, entry)

configFunc(label, 0.5, 1)
configFunc(entry, 1)

// But how do I add a widget now?
ctn.Add(label2) // This may refresh the view... 
configFunc(label2, 0.5) // and here too
```
It doesn't look very attractive to me.

Reply:
OK, I think we can adapt the `NewResponsive` function to make this:

```go

label := widget.NewLabel...
entry := widget.NewEntry...
other := widget.NewEntry...

config := []Ratios{
  {Object: label, Ratios: []float32{1, 0.5}}, // only the label
}

// use the config (or nil)
ctn := NewReponsive(&config, label, entry) // note that entry is not configured

// adding anthor widget
config.SetRatio(other,0.5, 0.5, 1) // if needed...
ctn.Add(other)
```

That means that I need to keep the `Ratios` configuration inside the layout, and to remove the `Responsive()` function + responsive widget.

Reply:
This seems cleaner